pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -464,6 +466,23 @@ func translateRoute(
 	if in.CorsPolicy != nil {
 		out.TypedPerFilterConfig[wellknown.CORS] = protoconv.MessageToAny(TranslateCORSPolicy(in.CorsPolicy))
 	}
+	if len(hostnames) > 0 {
+		for _, hostname := range hostnames {
+			statefulSvc := serviceRegistry[hostname]
+			if statefulConfig := util.MaybeBuildStatefulSessionFilterConfig(statefulSvc); statefulConfig != nil {
```

Comment:
A bit hard to read. A comment would be good. 

serviceRegistry[hostname] doesn't return a statefulSvc - but a svc which may be stateful or not. 

The MaybeBuild returns a config if the svc is stateful.